drivers/display/ssd1306.c

```diff
@@ -373,23 +376,36 @@ static int ssd1306_set_orientation(const struct device *dev,
 static int ssd1306_set_pixel_format(const struct device *dev,
 				    const enum display_pixel_format pf)
 {
-	if (pf == PIXEL_FORMAT_MONO10) {
-		return 0;
+	struct ssd1306_data *data = dev->data;
+
+	if (0U == (pf & (PIXEL_FORMAT_MONO10 | PIXEL_FORMAT_MONO01))) {
+		LOG_ERR("Unsupported");
+		return -ENOTSUP;
 	}
-	LOG_ERR("Unsupported");
-	return -ENOTSUP;
+	if (pf != data->current_format) {
+		data->current_format = pf;
+		uint8_t cmd_buf[] = {((pf == PIXEL_FORMAT_MONO10) ? SSD1306_SET_REVERSE_DISPLAY
+								  : SSD1306_SET_NORMAL_DISPLAY)};
+		if (ssd1306_write_bus(dev, cmd_buf, sizeof(cmd_buf), true)) {
+			return -EIO;
+		}
+	}
+	return 0;
 }
 
 static int ssd1306_init_device(const struct device *dev)
 {
 	const struct ssd1306_config *config = dev->config;
+	struct ssd1306_data *data = dev->data;
 
 	uint8_t cmd_buf[] = {
 		SSD1306_SET_ENTIRE_DISPLAY_OFF,
 		(config->color_inversion ? SSD1306_SET_REVERSE_DISPLAY
 					 : SSD1306_SET_NORMAL_DISPLAY),
 	};
 
+	data->current_format = config->color_inversion ? PIXEL_FORMAT_MONO10 : PIXEL_FORMAT_MONO01;
```

Comment:
Could also be done during instantiation, see ugly macro at the end of the file.

Reply:
I left it as it is for the moment. Like this, the data struct is uninitialized (as in pretty much every other module, I came across) and can live in the noinit section, instead of data. I prefer it this way, but if you have strong feelings about it, I can change it.